services/wallet/router/router.go

```diff
@@ -437,22 +277,41 @@ func newSuggestedRoutes(
 	}
 }
 
-func NewRouter(s *Service) *Router {
+func NewRouter(rpcClient *rpc.Client, transactor *transactions.Transactor, tokenManager *token.Manager, marketManager *market.Manager,
+	collectibles *collectibles.Service, collectiblesManager *collectibles.Manager, ensService *ens.Service, stickersService *stickers.Service) *Router {
 	bridges := make(map[string]bridge.Bridge)
-	transfer := bridge.NewTransferBridge(s.rpcClient, s.transactor)
-	erc721Transfer := bridge.NewERC721TransferBridge(s.rpcClient, s.transactor)
-	erc1155Transfer := bridge.NewERC1155TransferBridge(s.rpcClient, s.transactor)
-	cbridge := bridge.NewCbridge(s.rpcClient, s.transactor, s.tokenManager)
-	hop := bridge.NewHopBridge(s.rpcClient, s.transactor, s.tokenManager)
-	paraswap := bridge.NewSwapParaswap(s.rpcClient, s.transactor, s.tokenManager)
+	transfer := bridge.NewTransferBridge(rpcClient, transactor)
+	erc721Transfer := bridge.NewERC721TransferBridge(rpcClient, transactor)
+	erc1155Transfer := bridge.NewERC1155TransferBridge(rpcClient, transactor)
+	cbridge := bridge.NewCbridge(rpcClient, transactor, tokenManager)
+	hop := bridge.NewHopBridge(rpcClient, transactor, tokenManager)
+	paraswap := bridge.NewSwapParaswap(rpcClient, transactor, tokenManager)
```

Comment:
I'm sorry that I missed this previously. I notice that you've simplified the bridge inits by considerably increasing the `NewRouter()` init function's parameter complexity. I think that it is fine to pass in a single struct with multiple complex fields.

The Options pattern can help in this case or a config pattern. But also the original approach was fine. What problem does this change solve? Perhaps I'm not noticing something.

Reply:
@Samyoul That was an easy way to solve the cyclic dependency, that I ran into while moving the router to a new `router` package.

`router.go` is importing the package `wallet` and `wallet/api.go` is importing `router` package while it's part of `wallet` package, that results in cyclic dependency.

Reply:
Oh I see. Ok. Let's not spend too much brain power on this then. Thank you.